taskcluster/adhoc_taskgraph/transforms/mac_notarize.py

```diff
@@ -0,0 +1,45 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+"""
+Create signing tasks.
+"""
+from taskgraph.transforms.base import TransformSequence
+
+transforms = TransformSequence()
+
+
+@transforms.add
+def build_notarize_task(config, tasks):
+    for task in tasks:
+        if config.params["level"] != "3":
+            # Notarization only runs on level 3
+            continue
+        dep = task["primary-dependency"]
+        if not dep.attributes["manifest"].get("signingscript-notarization"):
+            continue
+        task["dependencies"] = {"signing": dep.label}
+        artifact_prefix = (
+            task["attributes"].get("artifact_prefix", "public").rstrip("/")
+        )
+        if not artifact_prefix.startswith("public"):
+            scopes = task.setdefault("scopes", [])
+            scopes.append(f"queue:get-artifact:{artifact_prefix}/*")
```

Comment:
This seems like something that could be more broadly added to all adhoc-signing tasks.